src/vm.rs

```diff
@@ -175,12 +179,37 @@ impl<VCpuType: VirtualCPU> UhyveVm<VCpuType> {
 		);
 	}
 
+	#[cfg(feature = "aslr")]
+	fn generate_start_address(&mut self, object_mem_size: u64) -> u64 {
+		let mut rng = rand::thread_rng();
+
+		// TODO: This breaks sometimes, causing the _kernel_ to return InsufficientMemory errors, despite the
+		// checks the boot process passing. Why?
+		//
+		// This seems to be a problem on the side of hermit-os/kernel, which does not behave very well if the
+		// start address is way too high. For some mysterious reason, this problem never occurs if we do not
+		// go past 0x3000000.
+		let start_address_upper_bound: u64 = std::cmp::min(
+			self.mem.memory_size as u64 - self.mem.guest_address.as_u64() - object_mem_size,
+			0x3000000,
+		);
+
+		// TODO: Add test. (from start_address_upper_bound-0x000001 to start_address_upper_bound+0x000001)
+		//
+		// We use 0x100000 as the offset for the start address so as to not use the zero page.
+		rng.gen_range(0x100000..start_address_upper_bound) & 0xffff_ffff_ffff_fff0
```

Comment:
This will not cut it for `aarch64`. I need to get Uhyve to run on `aarch64` and adjust this or make this feature explicitly `x86_64`-only in the meantime.